File: python/sanction_test.py
from onboarding_buddy_client.api.sanctions_api import SanctionsApi
from onboarding_buddy_client.models.individual_sanctions_check_request_m import IndividualSanctionsCheckRequestM
from onboarding_buddy_client.models.entity_sanctions_check_request_m import EntitySanctionsCheckRequestM
from onboarding_buddy_client.models.aircraft_sanctions_check_request_m import AircraftSanctionsCheckRequestM
from onboarding_buddy_client.models.vessel_sanctions_check_request_m import VesselSanctionsCheckRequestM
from onboarding_buddy_client.models.crypto_wallet_sanctions_check_request_m import CryptoWalletSanctionsCheckRequestM
from onboarding_buddy_client import Configuration, ApiClient
from dotenv import load_dotenv
import os

load_dotenv()

# Configure the client with authentication headers
config = Configuration(
    api_key={
        "AppKey": os.getenv("OB_APP_KEY"),
        "ApiKey": os.getenv("OB_API_KEY"),
        "ApiSecret": os.getenv("OB_API_SECRET")
    }
)

# Initialize the API client
api_client = ApiClient(configuration=config)
sanctions_api = SanctionsApi(api_client)

# Submit the email validation request
try:
    # Create an individual sanction request
    request = IndividualSanctionsCheckRequestM(
        first_name="YEVGENIY",
        last_name="PRIGOZHIN",
        birth_year="1961"
    )

    response = sanctions_api.individual(individual_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Full Name: {response.results[0].full_name}")
    print(f"First Name: {response.results[0].first_name}")
    print(f"Last Name: {response.results[0].last_name}")
    print(f"Program: {response.results[0].program}")
    print(f"Additional Information: {response.results[0].additional_info}")
    print(f"Date Of Birth: {response.results[0].date_of_birth}")
    print(f"Birth Year: {response.results[0].birth_year}")
    print(f"Nationality: {response.results[0].nationality}")
    print(f"Gender: {response.results[0].gender}")
    print(f"Linked To: {response.results[0].linked_to}")
    print(f"Secondary Sanctions Risk: {response.results[0].secondary_sanctions_risk}")
    
except Exception as e:
    print(f"Error: {e}")

# Submit the entity request
try:
    # Create an entity sanctions check request
    request = EntitySanctionsCheckRequestM(
        name="INTERNET RESEARCH AGENCY LLC"
    )
    
    response = sanctions_api.entity(entity_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Full Name: {response.match[0].full_name}")
    print(f"Program: {response.match[0].program}")
    print("Alias List:")
    for x in response.match[0].alias_list:
        print(x.full_name)

    print("\nAddress List:")        
    for y in response.match[0].address_list:
        print(f"{y.address} {y.city_state_province_postal_code} {y.country}")

    print("\nLinked Individuals:")        
    for z in response.match[0].linked_individuals:
        print(f"{z.full_name}")

    print(f"\nSecondary Sanctions Risk: {response.match[0].secondary_sanctions_risk}")
    
except Exception as e:
    print(f"Error: {e}")

# Submit the aircraft request
try:
    # Create an aircraft sanctions request
    request = AircraftSanctionsCheckRequestM(
        name="RA-02791"
    )

    response = sanctions_api.aircraft(aircraft_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Full Name: {response.match[0].full_name}")
    print(f"Program: {response.match[0].program}")
    print(f"Manufacture Date: {response.match[0].manufacture_date}")
    print(f"Model: {response.match[0].model}")
    print(f"Operator: {response.match[0].operator}")
    print(f"Mode S Transponder Code: {response.match[0].mode_s_transponder_code}")
    print(f"Serial Identification: {response.match[0].serial_identification}")
    print(f"Linked To: {response.match[0].linked_to}")
    print(f"Alias: {response.match[0].alias_list[0].full_name}")
    
except Exception as e:
    print(f"Error: {e}")

# Submit the vessel sanction request
try:
    # Create a vessel sanction check request
    request = VesselSanctionsCheckRequestM(
        name="HWANG GUM SAN 2"
    )

    response = sanctions_api.vessel(vessel_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Full Name: {response.match[0].full_name}")
    print(f"Program: {response.match[0].program}")
    print(f"Additional Info: {response.match[0].additional_info}")
    print(f"Secondary Sanctions Risk: {response.match[0].secondary_sanctions_risk}")
    print(f"Vessel Type: {response.match[0].vessel_type}")
    print(f"Vessel Flag: {response.match[0].vessel_flag}")
    
except Exception as e:
    print(f"Error: {e}")

# Submit the crypto wallet request
try:
    # Create an crypto wallet sanctions request
    request = CryptoWalletSanctionsCheckRequestM(
        address="0X098B716B8AAF21512996DC57EB0615E2383E2F96"
    )

    response = sanctions_api.crypto_wallet(crypto_wallet_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Entity Name: {response.entity_match[0].full_name}")
    print(f"Program: {response.entity_match[0].program}")
    print(f"Additional Info: {response.entity_match[0].additional_info}")
    print(f"Secondary Sanctions Risk: {response.entity_match[0].secondary_sanctions_risk}")

    print("\nAddress List:")        
    for x in response.entity_match[0].address_list:
        print(f"{x.address} {x.city_state_province_postal_code} {x.country}")

    print("\nCurrency List:")
    for y in response.entity_match[0].digital_currency_address:
        print(f"{y.instrument}: {y.address}")


    print("\nAlias List:")        
    for z in response.entity_match[0].also_known_as:
        print(f"{z}")

except Exception as e:
    print(f"Error: {e}")

File: typescript/src/sanction_test.ts
import OnboardingBuddyClient from '@onboardingbuddy/onboardingbuddy-client'
import * as dotenv from 'dotenv'
dotenv.config()

const obAppKey = process.env.OB_APP_KEY || '';
const obApiKey = process.env.OB_API_KEY || '';
const obApiSecret = process.env.OB_API_SECRET || '';

const client = new OnboardingBuddyClient(obAppKey, obApiKey, obApiSecret);

const individualSanctions = client.checkIndividualSanctions({
  firstName: 'YEVGENIY',
  lastName: 'PRIGOZHIN',
  birthYear: '1961',
});

individualSanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)

            if (response.data.results)
            {
                console.log(`Full Name: ${response.data.results[0].fullName}`)
                console.log(`First Name: ${response.data.results[0].firstName}`)
                console.log(`Last Name: ${response.data.results[0].lastName}`)
                console.log(`Program: ${response.data.results[0].program}`)
                console.log(`Additional Information: ${response.data.results[0].additionalInfo}`)
                console.log(`Date Of Birth: ${response.data.results[0].dateOfBirth}`)
                console.log(`Birth Year: ${response.data.results[0].birthYear}`)
                console.log(`Nationality: ${response.data.results[0].nationality}`)
                console.log(`Gender: ${response.data.results[0].gender}`)
                console.log(`Linked To: ${response.data.results[0].linkedTo}`)
                console.log(`Secondary Sanctions Risk: ${response.data.results[0].secondarySanctionsRisk}`)
            }
        })
    .catch(error=> console.log(error.response.data));

const entitySanctions = client.checkEntitySanctions({
  name: 'INTERNET RESEARCH AGENCY LLC'
});

entitySanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)
            if (response.data.match)
            {
                console.log(`Full Name: ${response.data.match[0].fullName}`)
                console.log(`Program: ${response.data.match[0].program}`)
                
                console.log(`\nAlias List:`)     
                
                if (response.data.match[0].aliasList)
                {
                    response.data.match[0].aliasList.forEach(x => {
                        console.log(x.fullName);
                    });
                }
                
                console.log(`\nAddress List:`)        
                if (response.data.match[0].addressList)
                {
                    response.data.match[0].addressList.forEach(y => {
                        console.log(`${y.address} ${y.cityStateProvincePostalCode} ${y.country}`);
                    });
                }
                
                console.log(`\nLinked Individuals:`)        
                if (response.data.match[0].linkedIndividuals)
                {
                    response.data.match[0].linkedIndividuals.forEach(z => {
                        console.log(z.fullName);
                    });
                }
    
                console.log(`\nSecondary Sanctions Risk: ${response.data.match[0].secondarySanctionsRisk}`)
            }
            
        })
    .catch(error=> console.log(error.response.data));

const aircraftSanctions = client.checkAircraftSanctions({
  name: 'RA-02791'
});

aircraftSanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)
            if (response.data.match)
            {
                console.log(`Full Name: ${response.data.match[0].fullName}`)
                console.log(`Program: ${response.data.match[0].program}`)
                console.log(`Manufacture Date: ${response.data.match[0].manufactureDate}`)
                console.log(`Model: ${response.data.match[0].model}`)
                console.log(`Operator: ${response.data.match[0].operator}`)
                console.log(`Mode S Transponder Code: ${response.data.match[0].modeSTransponderCode}`)
                console.log(`Serial Identification: ${response.data.match[0].serialIdentification}`)
                console.log(`Linked To: ${response.data.match[0].linkedTo}`)
                if (response.data.match[0].aliasList)
                {
                    console.log(`Alias: ${response.data.match[0].aliasList[0].fullName}`)
                }
            }
        })
    .catch(error=> console.log(error.response.data));

    const vesselSanctions = client.checkVesselSanctions({
  name: 'HWANG GUM SAN 2'
});

vesselSanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)
            if (response.data.match)
            {
                console.log(`Full Name: ${response.data.match[0].fullName}`)
                console.log(`Program: ${response.data.match[0].program}`)
                console.log(`Additional Info: ${response.data.match[0].additionalInfo}`)
                console.log(`Secondary Sanctions Risk: ${response.data.match[0].secondarySanctionsRisk}`)
                console.log(`Vessel Type: ${response.data.match[0].vesselType}`)
                console.log(`Vessel Flag: ${response.data.match[0].vesselFlag}`)
            }
        })
    .catch(error=> console.log(error.response.data));

    const cryptoWalletSanctions = client.checkCryptoWalletSanctions({
  address: '0X098B716B8AAF21512996DC57EB0615E2383E2F96'
});

cryptoWalletSanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)
            if (response.data.entityMatch)
            {
                console.log(`Entity Name: ${response.data.entityMatch[0].fullName}`)
                console.log(`Program: ${response.data.entityMatch[0].program}`)
                console.log(`Additional Info: ${response.data.entityMatch[0].additionalInfo}`)
                console.log(`Secondary Sanctions Risk: ${response.data.entityMatch[0].secondarySanctionsRisk}`)

                console.log(`\nAddress List:`)        
                if (response.data.entityMatch[0].addressList)
                {
                    response.data.entityMatch[0].addressList.forEach(x => {
                        console.log(`${x.address} ${x.cityStateProvincePostalCode} ${x.country}`);
                    });
                }
                
                console.log(`\nCurrency List:`)
                if (response.data.entityMatch[0].digitalCurrencyAddress)
                {
                    response.data.entityMatch[0].digitalCurrencyAddress.forEach(y => {
                        console.log(`${y.instrument}: ${y.address}`);
                    });
                }
                
                console.log(`\nAlias List:`)        
                if (response.data.entityMatch[0].alsoKnownAs)
                {
                    response.data.entityMatch[0].alsoKnownAs.forEach(z => {
                        console.log(z);
                    });
                }
            }
        })
    .catch(error=> console.log(error.response.data));
